// [[Reason: Function declarations are not fixed]] ////

//// [tests/cases/compiler/declarationEmitDefaultExportWithStaticAssignment.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -4,35 +4,66 @@
 export declare class Foo {
 }
 //# sourceMappingURL=foo.d.ts.map
 //// [index1.d.ts]
-declare function Example(): void;
-declare namespace Example {
-    var Foo: typeof import("./foo").Foo;
-}
-export default Example;
+export default function Example(): void;
 //# sourceMappingURL=index1.d.ts.map
 //// [index2.d.ts]
 import { Foo } from './foo';
 export { Foo };
-declare function Example(): void;
-declare namespace Example {
-    var Foo: typeof import("./foo").Foo;
-}
-export default Example;
+export default function Example(): void;
 //# sourceMappingURL=index2.d.ts.map
 //// [index3.d.ts]
 export declare class Bar {
 }
-declare function Example(): void;
-declare namespace Example {
-    var Bar: typeof import("./index3").Bar;
-}
-export default Example;
+export default function Example(): void;
 //# sourceMappingURL=index3.d.ts.map
 //// [index4.d.ts]
 export declare function C(): any;
-export declare namespace C {
-    var A: () => void;
\ No newline at end of file
-    var B: () => void;
-}
-//# sourceMappingURL=index4.d.ts.map
+//# sourceMappingURL=index4.d.ts.map
+/// [Errors] ////
+
+index1.ts(2,25): error TS9009: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+index2.ts(3,25): error TS9009: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+index3.ts(2,25): error TS9009: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+index4.ts(5,17): error TS9009: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+
+
+==== foo.ts (0 errors) ====
+    export class Foo {}
+    
+==== index1.ts (1 errors) ====
+    import {Foo} from './foo';
+    export default function Example(): void {}
+                            ~~~~~~~
+!!! error TS9009: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+    Example.Foo = Foo
+    
+==== index2.ts (1 errors) ====
+    import {Foo} from './foo';
+    export {Foo};
+    export default function Example(): void {}
+                            ~~~~~~~
+!!! error TS9009: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+    Example.Foo = Foo
+    
+==== index3.ts (1 errors) ====
+    export class Bar {}
+    export default function Example(): void {}
+                            ~~~~~~~
+!!! error TS9009: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+    
+    Example.Bar = Bar
+    
+==== index4.ts (1 errors) ====
+    function A() {  }
+    
+    function B() { }
+    
+    export function C(): any {
+                    ~
+!!! error TS9009: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+      return null;
+    }
+    
+    C.A = A;
+    C.B = B;
\ No newline at end of file
